test/e2e/network/dns.go

```diff
@@ -88,6 +90,38 @@ var _ = common.SIGDescribe("DNS", func() {
 		validateDNSResults(ctx, f, pod, append(wheezyFileNames, jessieFileNames...))
 	})
 
+	/*
+		Release: v1.30
+		Testname: DNS, cluster
+		Description: When a Pod is created, the container hostname should match with spec hostname after setting the hostNetwork to true
+	*/
+	framework.ConformanceIt("should resolve the hostname and hostNetwork for a Pod", func(ctx context.Context) {
+		ginkgo.By("Creating a pod by setting hostname")
+		execPod := e2epod.CreateExecPodOrFail(ctx, f.ClientSet, f.Namespace.Name, dnsTestPodHostName, nil)
+		cmd := "hostname"
+		execPod.Spec.Hostname = dnsTestPodHostName
+		framework.Logf("Setting up hostname to: %v", execPod.Spec.Hostname)
+		stdout, err := e2eoutput.RunHostCmd(execPod.Namespace, execPod.Name, cmd)
+		if err != nil {
+			framework.Logf("Error running hostname command: %v", err)
+		}
+		trimmed := strings.TrimSpace(stdout)
+		gomega.Expect(execPod.Spec.Hostname).NotTo(gomega.Equal(trimmed))
```

Comment:
I find this harder to read than 
```suggestion
		if execPod.Spec.Hostname != trimmed {
		   framework.Failf("The spec.hostname is not same as container hostname, expected to contain: %s, got: %s", execPod.Spec.Hostname, stdout)
		}
```

Reply:
it is also the other way around, it should fail if it does not match

Reply:
Done